transformer_engine/pytorch/attention.py

```diff
@@ -907,6 +903,11 @@ class DotProductAttention(torch.nn.Module):
     layer_number: int, default = `None`
                  layer number of the current `DotProductAttention` when multiple such modules
                  are concatenated, for instance in consecutive transformer blocks.
+    attn_mask_type: {'causal', 'padding', 'no_mask'}, default = `causal`
+                   type of attention mask passed into softmax operation. Overridden by
+                   :attr:`attn_mask_type` in the `forward` method. The forward
+                   arg is useful for dynamically changing mask types and the init arg
```

Comment:
Maybe add here an example where people would want that (training and inference)?